iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -290,8 +324,12 @@ void populateScalarizeMathOps(RewritePatternSet &patterns) {
       patterns.getContext());
 }
 
-std::unique_ptr<OperationPass<FuncOp>> createTestLLVMGPUScalarizeMathOpPass() {
-  return std::make_unique<TestLLVMGPUScalarizeMathOpPass>();
+void populateConvertSharedMemoryAllocOps(RewritePatternSet &patterns) {
+  patterns.add<ConvertSharedMemAllocOp>(patterns.getContext());
+}
+
+std::unique_ptr<OperationPass<FuncOp>> createTestLLVMGPULegalizePass() {
```

Comment:
This cannot be a `FuncOp` pass if you are adding a global variable. Can cause weird behavior.

Reply:
Good catch. Changed it to module pass.